sql/api/src/main/scala/org/apache/spark/sql/types/StructField.scala

```diff
@@ -150,10 +150,13 @@ case class StructField(
   /**
    * Return the default value of this StructField. This is used for storing the default value of a
    * function parameter.
+   *
+   * It is present when the field represents a function parameter with a default value, such as
+   * `CREATE FUNCTION f(arg INT DEFAULT 42) RETURN ...`.
    */
-  private[sql] def getDefault(): Option[String] = {
```

Comment:
Let's keep the original function name if possible? I think the docstring itself makes it clear it's for function parameter. It would be good to reduce cosmetic changes given the PR is already large.